pkg/logs/tailers/windowsevent/tailer.go

```diff
@@ -89,210 +94,257 @@ func (t *Tailer) Identifier() string {
 	return Identifier(t.config.ChannelPath, t.config.Query)
 }
 
-// toMessage converts an XML message into json
-func (t *Tailer) toMessage(re *richEvent) (*message.Message, error) { //nolint:unused
-	event := re.xmlEvent
-	log.Debug("Rendered XML:", event)
-	mxj.PrependAttrWithHyphen(false)
-	mv, err := mxj.NewMapXml([]byte(event))
+func (t *Tailer) toMessage(re *richEvent) (*message.Message, error) {
+	jsonEvent, err := eventToJSON(re)
 	if err != nil {
 		return &message.Message{}, err
 	}
+	return message.NewMessageWithSource([]byte(jsonEvent), message.StatusInfo, t.source, time.Now().UnixNano()), nil
+}
 
-	// extract then modify the Event.EventData.Data field to have a key value mapping
-	dataField, err := extractDataField(mv)
-	if err != nil {
-		log.Debugf("Error extracting data field: %s", err)
-	} else {
-		err = mv.SetValueForPath(dataField, dataPath)
-		if err != nil {
-			log.Debugf("Error formatting %s: %s", dataPath, err)
+// Start starts tailing the event log.
+func (t *Tailer) Start(bookmark string) {
+	log.Infof("Starting windows event log tailing for channel %s query %s", t.config.ChannelPath, t.config.Query)
+	t.stop = make(chan struct{})
+	t.done = make(chan struct{})
+	go t.forwardMessages()
+	t.decoder.Start()
+	go t.tail(bookmark)
+}
+
+// Stop stops the tailer
+func (t *Tailer) Stop() {
+	log.Info("Stop tailing windows event log")
+	close(t.stop)
+	<-t.done
+
+	t.decoder.Stop()
+
+	t.sub.Stop()
+}
+
+func (t *Tailer) forwardMessages() {
+	for decodedMessage := range t.decoder.OutputChan {
+		if len(decodedMessage.Content) > 0 {
+			t.outputChan <- decodedMessage
 		}
 	}
+}
 
-	// extract, parse then modify the Event.EventData.Binary data field
-	binaryData, err := extractParsedBinaryData(mv)
-	if err != nil {
-		log.Debugf("Error extracting binary data: %s", err)
+// tail subscribes to the channel for the windows events
+func (t *Tailer) tail(bookmark string) {
+	if t.evtapi == nil {
+		t.evtapi = winevtapi.New()
+	}
+
+	var err error
+
+	opts := []evtsubscribe.PullSubscriptionOption{
+		evtsubscribe.WithWindowsEventLogAPI(t.evtapi),
+		evtsubscribe.WithEventBatchCount(10),
+		evtsubscribe.WithNotifyEventsAvailable(),
+	}
+
+	if bookmark != "" {
+		// load bookmark
+		t.bookmark, err = evtbookmark.New(
+			evtbookmark.WithWindowsEventLogAPI(t.evtapi),
+			evtbookmark.FromXML(bookmark))
+		if err != nil {
+			err = fmt.Errorf("error creating bookmark: %v", err)
+			log.Errorf("%v", err)
+			t.source.Status.Error(err)
+			return
```

Comment:
Would it make sense to start from the current position, rather than giving up entirely?

Reply:
https://github.com/DataDog/datadog-agent/pull/19761/commits/ebaffbd430b4bcfee9ba8f47937961773812884f